app/presenters/formats/service_sign_in_presenter.rb

```diff
@@ -0,0 +1,133 @@
+module Formats
+  class ServiceSignInPresenter
+    attr_reader :content
+
+    def initialize(content)
+      @content = content.deep_symbolize_keys
+    end
+
+    def render_for_publishing_api
+      {
+        schema_name: "service_sign_in",
+        rendering_app: "government-frontend",
+        publishing_app: "publisher",
+        document_type: "service_sign_in",
+        locale: locale,
+        update_type: update_type,
+        change_note: change_note,
+        base_path: base_path,
+        routes: routes,
+        title: title,
+        description: description,
+        public_updated_at: public_updated_at,
+        details: details,
+      }
+    end
+
+    def content_id
+      begin
+        content_item["content_id"]
+      rescue GdsApi::ContentStore::ItemNotFound
+        SecureRandom.uuid
+      end
+    end
+
+    def links
+      {
+        parent: [parent.content_id]
+      }
+    end
+
+  private
+
+    def locale
+      content[:locale]
+    end
+
+    def update_type
+      content[:update_type]
+    end
+
+    def change_note
+      content[:change_note]
+    end
+
+    def base_path
+      "/#{parent_slug}/sign-in"
+    end
+
+    def routes
+      [
+        { path: base_path.to_s, type: "prefix" },
+      ]
+    end
+
+    def title
+      parent.title
+    end
+
+    def description
+      parent.overview
+    end
+
+    def public_updated_at
+      return DateTime.now.rfc3339 if update_type == "major"
+      content_item["public_updated_at"]
```

Comment:
Since this will only be run when there's an actual change, you could choose to always use the current time. That would probably prevent you from having to talk to publishing-api/content-store at all.

Reply:
These pages could be republished to fix a typo or change the grammar in the text. In that case, you want to publish it as a "minor" update and not want to change the "public_updated_at" date which is for major updates.

Reply:
I've checked with @kevindew, and he says that when you don't send `public_updated_at`, and the update type is `minor`, the `public_updated_at` in the publishing-api will not changed.

Reply:
Ok we should probably include this information in the Publishing API docs, because to me this reads as `public_updated_at` is required unless the document_type is "contact" or "government": https://github.com/alphagov/publishing-api/blob/master/doc/api.md#put-v2contentcontent_id

Reply:
Oh dear! yes we should fix that